packages/api-db-mongodb/src/db/invoice.ts

```diff
@@ -0,0 +1,205 @@
+import {Cursor} from './cursor'
+import {MaxResultsPerPage} from './defaults'
+
+export class MongoDBInvoiceAdapter implements DBInvoiceAdapter {
+  private invoices: Collection<DBInvoice>
+  private locale: string
+
+  constructor(db: Db, locale: string) {
+    this.invoices = db.collection(CollectionName.Invoices)
+    this.locale = locale
+  }
+
+  async createInvoice({input}: CreateInvoiceArgs): Promise<Invoice> {
+    const {ops} = await this.invoices.insertOne({
+      createdAt: new Date(),
+      modifiedAt: new Date(),
+      mail: input.mail,
+      userID: input.userID,
+      description: input.description,
+      payedAt: input.payedAt,
+      items: input.items
+    })
+
+    const {_id: id, ...invoice} = ops[0]
+    return {id, ...invoice}
+  }
+
+  async updateInvoice({id, input}: UpdateInvoiceArgs): Promise<OptionalInvoice> {
+    const {value} = await this.invoices.findOneAndUpdate(
+      {_id: id},
+      {
+        $set: {
+          modifiedAt: new Date(),
+          mail: input.mail,
+          userID: input.userID,
+          description: input.description,
+          payedAt: input.payedAt,
+          items: input.items
+        }
+      },
+      {returnOriginal: false}
+    )
+
+    if (!value) return null
+
+    const {_id: outID, ...invoice} = value
+    return {id: outID, ...invoice}
+  }
+
+  async deleteInvoice({id}: DeleteInvoiceArgs): Promise<string | null> {
+    const {deletedCount} = await this.invoices.deleteOne({_id: id})
+    return deletedCount !== 0 ? id : null
+  }
+
+  async getInvoicesByID(ids: readonly string[]): Promise<OptionalInvoice[]> {
+    const invoices = await this.invoices.find({_id: {$in: ids}}).toArray()
+    const invoiceMap = Object.fromEntries(
+      invoices.map(({_id: id, ...invoice}) => [id, {id, ...invoice}])
+    )
+
+    return ids.map(id => invoiceMap[id] ?? null)
+  }
+
+  async getInvoices({
+    filter,
+    sort,
+    order,
+    cursor,
+    limit
+  }: GetInvoicesArgs): Promise<ConnectionResult<Invoice>> {
+    const limitCount = Math.min(limit.count, MaxResultsPerPage)
+    const sortDirection = limit.type === LimitType.First ? order : -order
+
+    const cursorData = cursor.type !== InputCursorType.None ? Cursor.from(cursor.data) : undefined
+
+    const expr =
+      order === SortOrder.Ascending
+        ? cursor.type === InputCursorType.After
+          ? '$gt'
+          : '$lt'
+        : cursor.type === InputCursorType.After
+        ? '$lt'
+        : '$gt'
+
+    const sortField = invoiceSortFieldForSort(sort)
+    const cursorFilter = cursorData
+      ? {
+          $or: [
+            {[sortField]: {[expr]: cursorData.date}},
+            {_id: {[expr]: cursorData.id}, [sortField]: cursorData.date}
+          ]
+        }
+      : {}
+
+    let textFilter: FilterQuery<any> = {}
+
+    // TODO: Rename to search
+    if (filter?.mail !== undefined) {
+      textFilter['$or'] = [{mail: {$regex: filter.mail, $options: 'i'}}]
```

Comment:
my system suggests using: `textFilter.or` instead

Reply:
what about the dollar sign? @faebebin